### PR TITLE
Bugfix and new function

### PR DESCRIPTION
**Pull request by vartika271987**
Bugfix in water_price, new function on water usage for livestock, renaming water usage for cropsBugfix in water_price, a new function on waterusage for livestock, renaming water usage for crops

### CODE REVIEW

File: R/water_price.R [hunk 1]
```diff
@@ -28,19 +28,22 @@ water_price <- function(gdx, file=NULL, level="reg", index=FALSE, index_baseyear
 
   if(level=="cell"){
     water<- as.magpie(-oq_water_cell)
+    water<- as.magpie(round(water,digits))
```
[the remark this answers is not in the document]
Comment (vartika271987) [reply R/water_price.R]: Done

File: R/reportWaterUsageLivestock.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#'   \dontrun{
+#'     x <- reportWaterUsageLivestock(gdx)
+#'   }
+#' 
+
+reportWaterUsageLivestock<-function(gdx){
+  x <- water_usage(gdx,level="regglo",users="kli",sum=TRUE,digits=3)
+  getNames(x) <- "Resources|Water|Withdrawal|Agriculture|Livestock (km3/yr)"
+  return(x)
+}
```
[the remark this answers is not in the document]
Comment (vartika271987) [reply R/reportWaterUsageLivestock.R]: Removed and modified the base function that now reports water usage for agricultural sector, crops and livestock separately.

**Review by flohump — APPROVED** [review]
Seems fine to me.

File: R/reportWaterUsage.R [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#' 
+#'   \dontrun{
+#'     x <- reportWaterUsage(gdx)
+#'   }
+#' 
+
+
+reportWaterUsage<-function(gdx, detail=TRUE) {
+  
+  x <- water_usage(gdx,level="regglo",users=NULL,sum=TRUE,digits=3)[,,"agriculture"]
```
Comment (mishkos) [inline R/reportWaterUsage.R:20]: the sum has to be set to FALSE so that the sectors are not all summed up.

File: R/reportWaterUsage.R [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  getNames(x) <- "Resources|Water|Withdrawal|Agriculture (km3/yr)"
+  return(x)
```
Comment (mishkos) [inline R/reportWaterUsage.R:22]: The object has to be return at the end of the function, cause if it's returned here, the stuff in the if-condition can be left out.

File: R/reportWaterUsage.R [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  
+  if (detail==TRUE) {
+    out <-x
```
Comment (mishkos) [inline R/reportWaterUsage.R:25]: you don't need this assignment here. Instead just use `x` object at line 29. `out` is not necessary

**Review by mishkos — CHANGES_REQUESTED** [review]

File: R/reportWaterUsage.R [hunk 6]
```diff
@@ -1,20 +1,40 @@
...
+  if (detail==TRUE) {
+    y <- water_usage(gdx,level="regglo",users="kcr",sum=FALSE,digits=3)[,,"agriculture"]
+    getNames(y) <- "Resources|Water|Withdrawal|Agriculture|Crops (km3/yr)"
+    out <- mbind(out,y)
+    
+    z <- water_usage(gdx,level="regglo",users="kli",sum=FALSE,digits=3)[,,"agriculture"]
+    getNames(z) <- "Resources|Water|Withdrawal|Agriculture|Livestock (km3/yr)"
+    out <- mbind(out,z)
+    
+    return(out)
```
Comment (mishkos) [inline R/reportWaterUsage.R:32]: `return(out)` should be out of the if-condition brackets, to ensure retrung of total ag water values when `detail=F`

Comment (vartika271987) [reply R/reportWaterUsage.R]: You are right. Fixed now.

**Review by mishkos — APPROVED** [review]
all good

**Pull request merged**